app/src/main/java/org/groundplatform/android/ui/map/gms/features/LineStringRenderer.kt

```diff
@@ -72,6 +85,90 @@ constructor(private val resources: Resources, private val bitmapUtil: BitmapUtil
       jointType = JointType.ROUND
       zIndex = POLYLINE_Z
     }
+
+    updateTooltipMarker(map, geometry)
     return polyline
   }
+
+  private fun updateTooltipMarker(map: GoogleMap, geometry: LineString) {
+    if (geometry.coordinates.size < 2) {
```

Comment:
If the marker is generic, why does number of coordinates matter? Shouldn't it be removed if the tooltip is null or empty?